### PR TITLE
docs: Add squad field to onboarding form

### PR DESCRIPTION
**Pull request by arbrandes**
Adds an optional `squad` field to the onboarding ticket template.

Also rename the `push-pull-all` team to `2u-edx-legacy`.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/01-github-request---onboarding.yml [hunk 1]
```diff
@@ -26,6 +26,11 @@ body:
         * ...
     validations:
       required: true
+  - type: input
+    id: squad
+    attributes:
+      label: User Squad
+      description: If the users specified above are members of a named squad or team at your firm, optionally specify it here.  This will help simplify access management in the future.
```
Comment (arbrandes) [inline .github/ISSUE_TEMPLATE/01-github-request---onboarding.yml:33]: Should we make the distinction between squad and Github team?  Does it matter if folks specify a Github team, here?

Comment (sarina) [reply .github/ISSUE_TEMPLATE/01-github-request---onboarding.yml:33]: I don't know - probably a question for Ned

**Review by sarina — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/01-github-request---onboarding.yml [hunk 2]
```diff
@@ -49,7 +54,7 @@ body:
     id: requests
     attributes:
       label: Special Requests
-      description: For now, all new 2U/edX employees will be granted access to `push-pull-all`. If these users need to be added to any specific teams or given any elevated privileges, note them here. 
+      description: For now, all new 2U/edX employees will be granted access to `2u-edx-legacy`. If these users need to be added to any specific Github teams or given any elevated privileges, note them here.
```
Comment (nedbat) [inline .github/ISSUE_TEMPLATE/01-github-request---onboarding.yml:57]: I thought we weren't adding new people to 2u-edx-legacy?  I know there are newer engineers who are not in the team.

File: .github/ISSUE_TEMPLATE/01-github-request---onboarding.yml (hunk 1 above)
Comment (nedbat) [reply .github/ISSUE_TEMPLATE/01-github-request---onboarding.yml:33]: TBH, i don't know what information the 2U help desk will have when they fill this out.  This seems like a fine description to start with.

File: .github/ISSUE_TEMPLATE/01-github-request---onboarding.yml (hunk 2 above)
Comment (nedbat) [reply .github/ISSUE_TEMPLATE/01-github-request---onboarding.yml:57]: Hmm, maybe that was just a hiccup?  It looks like even newer people are in the team.

Comment (arbrandes) [reply .github/ISSUE_TEMPLATE/01-github-request---onboarding.yml:57]: As I understand it, we're still supposed to add people to it if requested.  What's fuzzy to me is if it's _not_ requested.

Comment (arbrandes) [reply .github/ISSUE_TEMPLATE/01-github-request---onboarding.yml:57]: I'm guessing we can deal with the wording here separately, so going ahead and merging the PR.

**Pull request merged**